lib/src/local_working_copy.rs

```diff
@@ -1238,8 +1239,7 @@ impl TreeState {
                 err: err.into(),
             })?;
         let size = conflict_data.len() as u64;
-        // TODO: Set the executable bit correctly (when possible) and preserve that on
-        // Windows like we do with the executable bit for regular files.
```

Comment:
I think the Windows issue still exist. The comment can be moved to `write_path_to_store()`?

Reply:
Agreed. And it's probably not very hard to fix it there if you're interested, but feel free to leave it for later.

Reply:
I see, the filemode issue doesn't repro on Windows so I didn't realise there was another one. I'll look deeper.

Reply:
ok, i think i understand this now - windows filesystems may not have `+x`, so we keep the last known value of the executable flag for ordinary files. i've done the same thing for conflicts:
```rs
                    #[cfg(windows)]
                    let executable = {
                        let () = executable; // use the variable
                        if let Some(merge) = current_tree_values.to_executable_merge() {
                            merge.into_resolved().unwrap_or(Some(false)).unwrap_or_default()
                        } else {
                            false
                        }
                    };
```